app/datatables/sample_datatable.rb

```diff
@@ -207,8 +218,12 @@ def get_raw_records
                           (sample_my_modules.my_module_id IN (#{ids.to_sql}) OR
                           sample_my_modules.id IS NULL))")
                           .references(:sample_my_modules)
+      if params[:assigned].present? && params[:assigned] == 'assigned'
+        samples = samples.joins('LEFT OUTER JOIN "my_modules" ON "my_modules"."id" = "sample_my_modules"."my_module_id"')
+                         .where('"my_modules"."experiment_id" = ?', @experiment.id)
+                         .where('"my_modules"."nr_of_assigned_samples" > 0')
+      end
     end
-
```

Comment:
I think you can move all three parts of the code here, as in: 

```
if params[:assigned] == 'assigned'
  samples = samples.where('"sample_my_modules"."id" > 0')
end
```
Correct me if I'm wrong...